pkg/dhcp6client/dhcp6client.go

```diff
@@ -134,12 +142,12 @@ func (c *Client) ReadReply() (*dhcp6.Packet, error) {
 
 			if udp.DestinationPort() == srcPort {
 				dhcp6p := &dhcp6.Packet{}
-				if err = dhcp6p.UnmarshalBinary(p.remaining()); err != nil {
+				// Not a valid DHCPv6 reply; keep listening.
```

Comment:
Comment is misplaced here, the line below can also produce a valid DHCP message.